rpc/jsonrpc/methods.go

```diff
@@ -1891,15 +1891,24 @@ func (s *Server) listSinceBlock(ctx context.Context, icmd interface{}) (interfac
 		return nil, errUnloadedWallet
 	}
 
-	tipHash, tipHeight := w.MainChainTip()
 	targetConf := int32(*cmd.TargetConfirmations)
 	if targetConf < 1 {
 		return nil, rpcErrorf(dcrjson.ErrRPCInvalidParameter, "target_confirmations must be positive")
 	}
 
+	tipHash, tipHeight := w.MainChainTip()
+	lastBlock := &tipHash
+	if targetConf > 0 {
+		var err error
+		lastBlock, err = w.FetchBlockHashForHeight((tipHeight + 1) - targetConf)
```

Comment:
this can be done without introducing a new wallet API by using the BlockInfo method with a BlockIdentifier created by a height.